### ISSUE TITLE
Unable to detect OpenJDK installed via Homebrew.

### ISSUE THREAD

**Issue by madelaney** [redhat-developer/vscode-java#2278]
While trying to load/use the Java Extension, using JDK's provided by [Homebrew](https://brew.sh/), I was unable to get the plugin to use the provided Homebrew OpenJDK installations.

##### Environment
- Operating System: Mac OS X (12.1)
- JDK version:  11.0.12 (2021-07-20)
- Visual Studio Code version: VSCodium 1.63.2
- Java extension version: v1.1.0

##### Steps To Reproduce
1. Install Java Extension plugin v1.1.0
2. Configure JDK's to point to Homebrew installed OpenJDK
2.1 JavaSE-1.11 `/usr/local/Cellar/openjdk@11/11.0.12`
2.2 JavaSE-1.8 `/usr/local/Cellar/openjdk@8/1.8.0+312`
3. Launch Maven Project
4. Observe the warning about no JDK11 found.

![Screen Shot 2022-01-10 at 8 32 33 AM](https://user-images.githubusercontent.com/6340271/148774975-eff135c5-5f3c-44ff-93d7-6903ad0a8b5f.png)

##### Current Result

The Java extensions fails to load.

##### Expected Result

I would expect the java extension to load the JDK11 installation provided by Homebrew.

##### Additional Informations

While trying to debug, I did try a few different versions of the `path` for `JavaSE-11`. I came across some older issues that suggested that due to Homebrew installing the JDK in a directory other then `/Library/Java/JavaVirtualMachines`, that the path should use the `libexec` directory (to avoid using symlinks). For example, don't use the path value of `/usr/local/Cellar/openjdk@11/11.0.12`, but instead use `/usr/local/Cellar/openjdk@11/11.0.12/libexec/openjdk.jdk/Contents/Home`. 

I did try this, and I didn't have any positive results from trying that different path.

My `java.configuration.runtimes` is configured as such:
```json
    "java.configuration.runtimes": [
      {
        "name": "JavaSE-1.8",
        "path": "/usr/local/Cellar/openjdk@8/1.8.0+312",
      },
      {
        "name": "JavaSE-11",
        "path": "/usr/local/Cellar/openjdk@11/11.0.12",
        "default": true
      }
    ],
    "java.trace.server": "verbose"
```

**Comment by Eskibear** [comment]
Try v1.2.0 ? 
https://github.com/redhat-developer/vscode-java/releases/tag/v1.2.0

**Comment by madelaney** [comment]
Sure, I can give that a shot.

Curious that I just installed the plugin, a few hours before creating the issue, but I didn't get the latest version. 💭

**Comment by madelaney** [comment]
Updating to v1.2.0 didn't resolve anything. However, setting `java.home` to the home of path of my OpenJDK 11 installation (`/usr/local/Cellar/openjdk@11/11.0.12`) seems to have resolved the issue.

In leu of the new information, I don't think it was the of `java.configuration.runtimes` but instead the lack of a `java.home` property. 

I find this kind of odd because I do have `JAVA_HOME` set in my terminal:
```
echo $JAVA_HOME
/usr/local/Cellar/openjdk@11/11.0.12

file ${JAVA_HOME}
/usr/local/Cellar/openjdk@11/11.0.12: directory
```

Maybe my shell `TCSH` is causing some issues?

**Comment by Eskibear** [comment]
`java.home` is deprecated since v1.2.0, where we embed a platform-specific JDK 17 inside to launch our tools. See https://github.com/redhat-developer/vscode-java/issues/2109 I'm also curious why it didn't work. 

How did you install v1.2.0? Marketplace is supposed to deliver corresponding package, if you download vsix from Github, for Mac OS X with Intel chips, you should choose `redhat.java-1.2.0@darwin-x64.vsix`. 

And I don't think `TCSH` have anything to do with this.

**Comment by madelaney** [comment]
@Eskibear, The plug was installed from the Marketplace.

**Comment by Eskibear** [comment]
We'll support to detect JDKs installed by Homebrew in coming v1.3.0. See https://github.com/redhat-developer/vscode-java/pull/2258

Currently to test if your case will be covered, you can run `npx jdk-utils` to list detected JDKs if you have [NodeJS](https://nodejs.org/) runtime installed.

**Comment by Eskibear** [comment]
fixed since v1.3.0, shall we close this one?
